internal/providers/azure/azure.go

```diff
@@ -100,62 +107,7 @@ func FetchFromOvfDevice(f *resource.Fetcher, ovfFsTypes []string) (types.Config,
 	logger.Debug("reading config")
 	rawConfig, err := ioutil.ReadFile(filepath.Join(mnt, configPath))
 	if err != nil && !os.IsNotExist(err) {
-		return types.Config{}, report.Report{}, fmt.Errorf("failed to read config: %v", err)
-	}
-
-	return util.ParseConfig(logger, rawConfig)
-}
-
-func waitForCdrom(logger *log.Logger, devicePath string) {
-	for !isCdromPresent(logger, devicePath) {
-		time.Sleep(time.Second)
-	}
-}
-
-func isCdromPresent(logger *log.Logger, devicePath string) bool {
```

Comment:
On second thought, we should probably retain part of this code, to verify that the specified device is a CD-ROM device.  At the moment, this PR will accept a non-CD-ROM disk device that happens to have a UDF filesystem on it.  If a user had the ability to bind a custom disk to a VM, they could use that to insert their own Ignition config, which seems bad.